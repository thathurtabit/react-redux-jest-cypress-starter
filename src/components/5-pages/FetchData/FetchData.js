import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import FetchDataStyled from './FetchData.styled';

const mapStateToProps = state => ({
  response: state.data.response,
});

const FetchData = ({ response }) => (
  <FetchDataStyled>{response}</FetchDataStyled>
);

export default connect(mapStateToProps)(FetchData);

FetchData.propTypes = {
  response: PropTypes.arrayOf(PropTypes.string),
};

FetchData.defaultProps = {
  response: [],
};
