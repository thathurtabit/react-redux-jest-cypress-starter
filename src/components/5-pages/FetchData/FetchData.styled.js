import styled from 'styled-components';

const FetchDataStyled = styled.section`
  position: 'relative';
`;

export default FetchDataStyled;
