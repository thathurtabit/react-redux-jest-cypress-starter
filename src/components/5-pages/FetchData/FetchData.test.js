import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { shallowWithTheme } from '../../../utils/test_config/testHelpers';

import FetchData from './FetchData';

const Title = 'Test Subtitle';

describe('FetchData', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(
      <FetchData response={[Title, Title, Title]} />
    );
    expect(shallowToJson(element)).toMatchSnapshot();
  });
});
