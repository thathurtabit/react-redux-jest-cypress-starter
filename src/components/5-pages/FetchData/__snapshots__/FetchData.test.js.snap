// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchData should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
